baselines/FedStar/FedStar/server.py

```diff
@@ -0,0 +1,304 @@
+
+
+def set_gpu_limits(gpu_id, gpu_memory):
+    os.environ["CUDA_VISIBLE_DEVICES"] = gpu_id
+    import tensorflow as tf
+
+    gpus = tf.config.list_physical_devices("GPU")
+    if not gpus:
+        print("No GPU's available. Server will run on CPU.")
+    else:
+        try:
+            tf.config.experimental.set_virtual_device_configuration(
+                gpus[0],
+                [
+                    tf.config.experimental.VirtualDeviceConfiguration(
+                        memory_limit=gpu_memory
+                    )
+                ],
+            )
+        except RuntimeError as e:
+            print(e)
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Flower Federated Learning - Server")
+    parser.add_argument(
+        "--server_address",
+        type=str,
+        default="[::]:10001",
+        required=False,
+        help=f"gRPC server address (default: [::]:8080)",
+    )
+    parser.add_argument(
+        "--rounds",
+        type=int,
+        default=100,
+        required=False,
+        help="Number of rounds of federated learning (default: 3)",
+    )
+    parser.add_argument(
+        "--batch_size",
+        type=int,
+        default=64,
+        required=False,
+        help="Training Batch Size (no default)",
+    )
+    parser.add_argument(
+        "--dataset_name",
+        type=str,
+        default=None,
+        required=False,
+        help="Relevant path to dataset (no default)",
+    )
+    parser.add_argument(
+        "--learning_rate",
+        type=float,
+        default=0.001,
+        required=False,
+        help="Model Learning Rate (no default)",
+    )
+    parser.add_argument(
+        "--num_clients",
+        type=int,
+        default=1,
+        required=False,
+        help="Minimum number of available clients required for sampling (default: 1)",
+    )
+    parser.add_argument(
+        "--min_sample_size",
+        type=int,
+        default=1,
+        required=False,
+        help="Minimum number of clients used for fit/evaluate (default: 1)",
+    )
+    parser.add_argument(
+        "--train_epochs",
+        type=int,
+        default=1,
+        required=False,
+        help="Clients train epochs (no default)",
+    )
+    parser.add_argument(
+        "--seed",
+        type=int,
+        default=2021,
+        required=False,
+        help="Seed for clients data shuffling (default: 2021)",
+    )
+    parser.add_argument(
+        "--gpu_memory",
+        type=int,
+        default=384,
+        required=False,
+        help="GPU Available Memory (no default)",
+    )
+    parser.add_argument(
+        "--eval_step",
+        type=int,
+        default=1,
+        required=False,
+        help="Evaluation Perform Step",
+    )
+    parser.add_argument(
+        "--verbose",
+        type=int,
+        default=1,
+        required=False,
+        help="Evaluation Detail to be displayed",
+    )
+    args = parser.parse_args()
```

Comment:
Flower baselines need to use `Hydra` configs. Please take a look at the example [fedprox]() baseline to get a quick taste on how `Hydra` works. There are many resources online as well. I even made a [short video](https://www.youtube.com/watch?v=zYT_buEw0iE&list=PLNG4feLHqCWlnj8a_E1A_n5zr2-8pafTB&index=9) recently as part as a tutorial series. 

Please don't use `argparse`.

Reply:
Thanks for sharing the resources, I have incorporated the hydra for server and client